client/AI/AIClientApp.cpp

```
@@ -199,119 +199,112 @@ void AIClientApp::HandleMessage(const Message& msg) {
         if (text.empty()) {
             ErrorLogger() << "AIClientApp::HandleMessage for HOST_ID : Got empty message text?!";
         } else {
-            try {
-                host_id = boost::lexical_cast<int>(text);
-            } catch (...) {
-                ErrorLogger() << "AIClientApp::HandleMessage for HOST_ID : Couldn't parese message text: " << text;
-            }
+            host_id = boost::lexical_cast<int>(text);
         }
```

Removing the empty catch `(...)` here is good, but not handling the `boost::bad_lexical_cast` that you know is possible is not.

Add a try and catch the `boost::bad_lexical_cast` exception with an error message.  You are expecting `text` to come from the server, but it is coming from outside of the application.